third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/Utility.cpp

```diff
@@ -139,6 +139,22 @@ Value createElectPredicateWarp0(Location loc, RewriterBase &rewriter) {
   return b.and_(warp0, createElectPredicate(loc, rewriter));
 }
 
+Value createTMAMulticastMask(Location loc, ConversionPatternRewriter &rewriter,
+                             uint16_t broadcastBits, int numCTAs) {
+  int blockBits = llvm::Log2_32(numCTAs);
+  uint16_t fixedBits = (~broadcastBits) & (numCTAs - 1);
+  uint32_t pattern = 1;
+  for (int i = 0; i < blockBits; ++i) {
+    if ((fixedBits & (1u << i)) == 0)
+      pattern |= (pattern << (1u << i));
+  }
+  auto b = TritonLLVMOpBuilder(loc, rewriter);
+  auto ctaId = nvgpu::ClusterCTAIdOp::create(rewriter, loc);
+  Value ctaId16 = b.trunc(rewriter.getIntegerType(16), ctaId);
+  Value base = b.and_(ctaId16, b.i16_val(fixedBits));
+  return b.shl(b.i16_val(static_cast<uint16_t>(pattern)), base);
```

Comment:
NIT: I would guess the codegen would be better if you truncate to 16 bits at the end, since PTX doesn't have many native 16-bit operations.